docs/analytics.md

```diff
@@ -0,0 +1,124 @@
+# Construct Hub Web Application Analytics
+
+The construct hub web application is designed to help users find and interact with construct libraries.
+
+In order for its operators to understand whether it indeed serves its purpose, the application offers an analytics
+mechanism to extract insights from the traffic it sees.
+
+This document describes that mechanism, detailing how, why, and what can be recorded.
+
+## Declaration of Intent
+
+As a general tenet, we are not interested in identifying individual user traffic. That is, it will not be possible to say which user
+searched or interacted with which construct. All we are interested in is aggregative stats to help improve the experience as a whole.
+
+## Recorded Data
+
+To understand which data should be recorded, we work backwards from the questions we'd like to ask it.
+For each question, we explain why are we asking it, and which metrics are needed in order to answer it.
+
+> Metrics are denoted using the common dot notation. It merely serves as a concise way to refer to measurements,
+> and does not infer any technical meaning. 
+
+### Are users satisfied with the hub?
+
+**Why:** The overall experience with the hub should be positive.
+
+To answer this, we need to understand whether existing users are abandoning the hub, or keep using it.
+Normally this is done by making sure that the amount of daily active users doesn't decrease over time.
+Since we don't identify individual users, we approximate by looking at the total number of distinct visits to the hub, assuming
+that individual user traffic doesn't have a large variance.
+
+**Required metrics:**
+
+- `visits.hub`: Number of distinct visits to the hub. (main page + package pages)
```

Comment:
I'd love a little summary section for `all required metrics` that shows them all in place just so we can easily look through them and compare.